src/awkward/_v2/cpp-headers/awkward/GrowableBuffer.h

```diff
@@ -302,7 +334,7 @@ namespace awkward {
     void
     append(PRIMITIVE datum) {
       if (ptr_->current_length() == ptr_->reserved()) {
-        add_panel((size_t)ceil(ptr_->reserved() * 1.5));
+        add_panel((size_t)ceil(ptr_->reserved() * options_.resize()));
```

Comment:
So right now, the algorithm is that each panel is a constant factor larger than the previous, so they'll grow exponentially. That made sense for the replacing-panel algorithm (so that there are logarithmically many replacements), but for the new discontiguous-panel algorithm, we want the panel size to grow less rapidly.

I think the best algorithm will probably be:

   1. Initial panel size is small, like 1024.
   2. Subsequent panel size (second onward) is large, like 1024*1024.

The actual values of the parameters (`initial` and `subsequent`) would be determined by performance tests: the criteria are (1) `initial` should be small enough that we can make a lot of them without running out of memory (~1 GB) and large enough that small arrays like unit tests don't trigger a second panel, and (2) `subsequent` should be large enough that << 1% of the `GrowableBuffer::append` time is spent managing panels; >> 99% is spent just copying data.

Reply:
Okay, good: so for now the "exponential growth" is a constant ($\exp(t)$ is an exponential!). But the thing to get addressed later is that it's the wrong algorithm. For later, anyway.